src/services/pending-queue.ts

```diff
@@ -169,6 +203,53 @@ export async function deletePendingQueueSubmission(pendingTimestamp: string) {
   await removePendingQueueSubmission(pendingTimestamp, 'DELETION')
 }
 
+/**
+ * Edit a PendingFormSubmission before it is processed based on the
+ * `pendingTimestamp` property. The function removes the submission from the
+ * pending queue and returns a prefill Id
+ *
+ * ### Example
+ *
+ * ```js
+ * const pendingTimestamp = '2020-07-29T01:03:26.573Z'
+ * const { preFillFormDataId, formId } =
+ *   await submissionService.editPendingQueueSubmission(pendingTimestamp)
+ * window.location.href = `https://mycoolforms.apps.oneblink.io/forms/${formId}?preFillFormDataId=${preFillFormDataId}`
+ * ```
+ *
+ * @param pendingTimestamp
+ */
+export async function editPendingQueueSubmission(
+  pendingTimestamp: string,
+): Promise<{ preFillFormDataId: string; formId: number }> {
+  try {
+    const formSubmission = await getFormSubmission(pendingTimestamp)
+    if (!formSubmission) {
+      throw new Error('Could not find formSubmision to edit')
+    }
+    await updatePendingQueueSubmission(
+      pendingTimestamp,
+      {
+        ...formSubmission,
+        pendingTimestamp,
+        isEditing: true,
+        isSubmitting: false,
+        error: undefined,
+      },
+      'EDIT_STARTED',
+      //this is to avoid logging to sentry twice
+      true,
+    )
+    const preFillFormDataId = `PENDING_SUBMISSION_${pendingTimestamp}`
+    const key = getPrefillKey(preFillFormDataId)
+    await utilsService.setLocalForageItem(key, formSubmission.submission)
```

Comment:
Should do this first incase it fails to avoid the pending queue item being marked as `isEditing`.

Reply:
Not sure I follow, let's chat post deployment